### ISSUE TITLE
Instead of no-extra-parens, I'd like to enforce extra parens

### ISSUE THREAD

**Issue by Sawtaytoes** [eslint/eslint#12969]
**What rule do you want to change?**
`no-extra-parens`

**Does this change cause the rule to produce more or fewer warnings?**
The opposite warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**
It's possible this requires a completely different rule identity, possibly a rename to `extra-parens`.

Right now, the rule is a negative, which is fine if I don't want parens, but instead, I want to actually enforce the use of extra parens as they assist in keeping consistent tabbing (no tangling tabs).

**Please provide some example code that this change will affect:**

```js
let func

// I want this to error or auto-fix
func = ({
  hello,
  someone,
}) =>
  someone
  ? someone
  : hello

// This should pass
func = ({
  hello,
  someone,
}) => (
  someone
  ? someone
  : hello
)

let test

// I want this to error or auto-fix
test = func()
  ? blah
  ? 'phil'
  : 'kevin'
  : 'jacob'

// This should pass
test = (
  func()
  ? (
    blah
    ? 'phil'
    : 'kevin'
  )
  : 'jacob'
)

let blah

// I want this to error or auto-fix
blah =
  someReallyLongFunctionName()

// This should pass
blah = (
  someReallyLongFunctionName()
)

func
test
blah
```

**What does the rule currently do for this code?**
Currently, the rule will either error if I use parens, or do nothing at all. There's no way to enforce parens for implicitly returning arrow functions or around nested ternaries.

**What will the rule do after it's changed?**
Depending on the configuration, it will either error on parens (and auto-fix), or it will require parens (and auto-fix).

**Are you willing to submit a pull request to implement this change?**
Yes.

**Comment by Sawtaytoes** [comment]
I found a library which does what I'm wanting, but only for functions, not assignments:
https://github.com/BerkeleyTrue/eslint-plugin-arrow-body-parens

Docs:
https://github.com/BerkeleyTrue/eslint-plugin-arrow-body-parens/blob/master/docs/lib/rules/parens.md

This repo hasn't been updated since 2017 though, but I'm wondering if the code could be used in the official ESLint repo with attribution. It's ISC licensed so that should be doable right?

I also went ahead and forked this for my own uses. I updated all packages, fixed the bugs, and got all tests to pass:
https://github.com/Sawtaytoes/Ghadyani-ESLint/tree/master/packages/arrow-body-parens

**Comment by ljharb** [comment]
Are you thinking of https://eslint.org/docs/rules/no-confusing-arrow ?

**Comment by Sawtaytoes** [comment]
Aside from the confusing name, the functionality is similar, but not the same.

The goal I have is to add newlines around the function body and have parens defining that behavior.

Right now, `no-confusing-arrow` has:
```js
fix(fixer) {
    // if `allowParens` is not set to true dont bother wrapping in parens
    return allowParens && fixer.replaceText(node.body, `(${sourceCode.getText(node.body)})`);
}
```

What I want is:
```js
fix(fixer) {
    if (!allowParens) {
        return;
    }
    
    if (enforceNewlines) {
        fixer.replaceText(node.body, `(\n${sourceCode.getText(node.body)}\n)`);
    }
    else {
        fixer.replaceText(node.body, `(${sourceCode.getText(node.body)})`);
    }
}
```

**Comment by kaicataldo** [comment]
You've listed a few cases in which you would like extra parens, but could you explain the heuristic you think should apply and how configuration would work?

**Comment by anikethsaha** [comment]
Is it something like `never` options for most of other rules ?

- `all`: disallows having extra parens
- `never`: disallows having no extra parents 

in simple words , it checks for parens around common statements like conditions, returns etc..

**Comment by kaicataldo** [comment]
I think we still need to understand what heuristic we're using to apply extra parens, or all parens would have to be doubled 😄
